### PR TITLE
Fixed handling of upstream disconnects.

### PR DESCRIPTION
**Pull request by adamshapiro0**
# Changes
- Increased the socket receive timeout to better allow for variable data latency
- Increased the default connection timeout in the C++ client

# Fixes
- Fixed handling of upstream socket disconnects in POSIX (behavioral difference between POSIX and FreeRTOS)

### CODE REVIEW

File: c/src/point_one/polaris/polaris.c [hunk 1]
```diff
@@ -490,21 +490,41 @@ int Polaris_Work(PolarisContext_t* context) {
...
 
+#ifdef P1_FREERTOS
+  // FreeRTOS does not use errno for recv() errors, but instead returns the
+  // equivalent error value. On a socket timeout, it returns 0 instead of
+  // ETIMEDOUT.
+  //
+  // We try to mimick the actual POSIX behavior to simplify the code below.
+  if (bytes_read == 0) {
+    bytes_read = -1;
+    errno = ETIMEDOUT;
```
Comment (anathan) [inline c/src/point_one/polaris/polaris.c:501]: Is errno defined in the freeertos context?

**Review by anathan — APPROVED** [review]
Minor question about errno, else looks fine.

File: c/src/point_one/polaris/polaris.c (hunk 1 above)
Comment (adamshapiro0) [reply c/src/point_one/polaris/polaris.c:501]: Yes, I believe so. We may need to include `#include "FreeRTOS_POSIX/errno.h"` actually - will need to check quickly

Comment (adamshapiro0) [reply c/src/point_one/polaris/polaris.c:501]: It appears to compile. It's bringing in `errno.h` from the GCC toolchain (`arm-none-eabi/include/sys/errno.h`). The FreeRTOS definitions from `projdefs.h` (e.g., `pdFREERTOS_ERRNO_ETIMEDOUT`) are intentionally defined to match the POSIX values, and all the relevant values appear to match `errno.h` as expected.

**Pull request merged**